reactioncommerce/reaction
# Add default parcel size setting

Resolves [#3019](https://github.com/reactioncommerce/reaction/issues/3019)
Add a settings panel within the shipping dashboard that permits defining default parcel sizes for products and variants that don't have dimensions defined. Also include sensible defaults for the default parcel size so that it's never null which could cause Shippo or other shipment providers to fail to calculate shipping correctly.

###### Fix
- Create component and container for new setting
- Create new package `reaction-shipping-parcel-size` to manage parcel size setting
- Add default values. Reference: [USPS](https://www.usps.com/ship/priority-mail.htm) 
- Update `Shops.json` with default parcel size values
- Update Shops and Product Variant schemas
- Update core translation
- Write unit tests for new setting
- Update `cart-create.app.test`

Test
- Reset and start reaction
- Run `meteor mongo` and 'use meteor in terminal (or start Robomongo)
- `db.Shops.find({}, {"defaultParcelSize": 1, "_id": 0})` should give `{ "defaultParcelSize" : { "weight" : 7, "length" : 11.25, "width" : 8.75, "height" : 6 } }`
- Login as admin and enable payment
- Go to shipping setting
- Enable Shippo (refresh methods if no values are shown)
- Check values shown in `Default parcel size` form fields. They should be the same values in the database
- Add a product
- Check the product dimensions. They should be the same values in the database
- Publish product, add to cart, and checkout (Shipping rates should be displayed)
- Check `Orders` in database. `items.parcel` should be the same value of product dimensions
- Change product dimensions and repeat step 10. 
- Check `Orders` in database and verify that `items.parcel` is equal to the new dimensions saved for the product
- Create another product and add a variant option. Variant dimensions should be equal to the values in the database
- Publish product, add variant option to cart, and checkout
- Check Orders in database. `items.parcel` should be the same value of variant option dimensions
- Change variant option dimensions and repeat steps 15 - 16
- Add a new product and edit dimension values
- Repeat steps 10 - 13
- Add a new product
- Repeat step 18 with zero dimension values (i.e: weight = 0, length = 0, width = 0, height = 0)
- Check `Orders` in database and verify that `items.parcel` is equal to the value for default parcel size
- Remove defaultParcelSize value in `Shops.json`
- Reset and run reaction
- Repeat above steps
